### ISSUE TITLE
AttributeError: 'BinaryClassificationEvaluator_wandb' object has no attribute 'truncate_dim'

### ISSUE THREAD

**Issue by danzzzlll** [dell-research-harvard/linktransformer#17]
Hello!
I was looking for frameworks to solve my problem and came across yours. I found it very interesting. Great job!

But while I trying to run the notebook “https://colab.research.google.com/drive/1tHitPGjMMI2Nvh4wwA8rdcbYfbLaJDvg?usp=sharing” from your tutorial I ran into an error with "label_col_name" (block "You can also use a dataset of positive and negatives")

AttributeError                            Traceback (most recent call last)
[<ipython-input-10-b780190f5909>](https://localhost:8080/#) in <cell line: 2>()
      1 # Call the train_model function
----> 2 saved_model_path = lt.train_model(
      3     model_path="oshizo/sbert-jsnli-luke-japanese-base-lite",
      4     data=df,
      5     left_col_names=["source_firm_title","source_address",

3 frames
[/usr/local/lib/python3.10/dist-packages/linktransformer/train_model.py](https://localhost:8080/#) in train_model(data, train_data, val_data, test_data, model_path, left_col_names, right_col_names, left_id_name, right_id_name, label_col_name, clus_id_col_name, clus_text_col_names, config_path, training_args, log_wandb)
    215 
    216     print("Training")
--> 217     best_model_path = train_biencoder(
    218         train_data=train_data,
    219         dev_data=val_data,

[/usr/local/lib/python3.10/dist-packages/linktransformer/modified_sbert/train.py](https://localhost:8080/#) in train_biencoder(train_data, dev_data, test_data, base_model, add_pooling_layer, train_batch_size, num_epochs, warm_up_perc, optimizer_params, loss_params, model_save_path, wandb_names, eval_steps_perc, eval_type, opt_model_description, opt_model_lang, loss_type)
    141 
    142     logger.info("Evaluate model without training")
--> 143     seq_evaluator(model, epoch=0, steps=0, output_path=model_save_path)
    144 
    145     # Train the model

[/usr/local/lib/python3.10/dist-packages/sentence_transformers/evaluation/SequentialEvaluator.py](https://localhost:8080/#) in __call__(self, model, output_path, epoch, steps)
     19         scores = []
     20         for evaluator in self.evaluators:
---> 21             scores.append(evaluator(model, output_path, epoch, steps))
     22 
     23         return self.main_score_function(scores)

[/usr/local/lib/python3.10/dist-packages/sentence_transformers/evaluation/BinaryClassificationEvaluator.py](https://localhost:8080/#) in __call__(self, model, output_path, epoch, steps)
    121         else:
    122             out_txt = ""
--> 123         if self.truncate_dim is not None:
    124             out_txt += f" (truncated to {self.truncate_dim})"
    125 

AttributeError: 'BinaryClassificationEvaluator_wandb' object has no attribute 'truncate_dim'

Can you please tell how to fix this error?

**Comment by econabhishek** [comment]
Thanks for pointing this out, @danzzzlll !  
It appears that the maintainers of sentence-transformers have slightly updated the evaluators. This could have been avoided if we had specified versions in the package dependency for the install - but we decided not to since this framework is still being actively maintained. I can fix this in the next version of the package. 

We'll either bring it to the current version of sentence-transformers or specify the version in our dependency. 

In the meantime, I would suggest downgrading sentence-transformers to 2.3.1  (a quick and dirty solution for now)- 
!pip install 'sentence-transformers==2.3.1' --force-reinstall


If you are doing this in colab, this may ask you to restart runtime - do that. But when you do it, don't reinstall the linktransformer package /the install command for sentence-transformers- it will be retained along with the downgraded sentence-transformers version. Simple run the import statements and the rest of the code. 

I will keep this issue open until we have fixed it.
